code.js

```diff
@@ -40,6 +41,7 @@ function heldKarp(cities, start, distance_matrix, memo){
         var city_minus_start = cities.filter(element => element !== start);
 
         for (var i = 0; i < cities.length; i++){
+            //here i pick a new start city that is not the old start city.
```

Comment:
Following you so far. My point is: are you really exploring all permutations of cities? That's what the brute-force approach would do, but is that also the case with memoization that avoids repeated work?